src/cross_module_fn.c

```diff
@@ -13,372 +13,95 @@
-{
-	PG_RETURN_DATUM(ts_cm_functions->remote_txn_id_in(fcinfo));
-}
-
-Datum
-ts_remote_txn_id_out(PG_FUNCTION_ARGS)
-{
-	PG_RETURN_DATUM(ts_cm_functions->remote_txn_id_out(fcinfo));
-}
-
-Datum
-ts_remote_txn_heal_data_node(PG_FUNCTION_ARGS)
-{
-	PG_RETURN_DATUM(ts_cm_functions->remote_txn_heal_data_node(fcinfo));
-}
-
-Datum
-ts_remote_connection_cache_show(PG_FUNCTION_ARGS)
-{
-	PG_RETURN_DATUM(ts_cm_functions->remote_connection_cache_show(fcinfo));
-}
+#define TSL_FORWARD_CALL(func)                                                                     \
+	TS_FUNCTION_INFO_V1(ts_##func);                                                                \
+	Datum ts_##func(PG_FUNCTION_ARGS)                                           \
+	{                                                                                              \
+		PG_RETURN_DATUM(ts_cm_functions->func(fcinfo));                                            \
+	}
+
+/* bgw policy functions */
+TSL_FORWARD_CALL(add_compress_chunks_policy);
+TSL_FORWARD_CALL(add_drop_chunks_policy);
+TSL_FORWARD_CALL(add_reorder_policy);
+TSL_FORWARD_CALL(alter_job_schedule);
+TSL_FORWARD_CALL(move_chunk);
+TSL_FORWARD_CALL(remove_compress_chunks_policy);
+TSL_FORWARD_CALL(remove_drop_chunks_policy);
+TSL_FORWARD_CALL(remove_reorder_policy);
+TSL_FORWARD_CALL(reorder_chunk);
+
+TSL_FORWARD_CALL(continuous_agg_invalidation_trigger);
+
+TSL_FORWARD_CALL(data_node_ping);
+TSL_FORWARD_CALL(data_node_block_new_chunks);
+TSL_FORWARD_CALL(data_node_allow_new_chunks);
+TSL_FORWARD_CALL(data_node_add);
+TSL_FORWARD_CALL(data_node_delete);
+TSL_FORWARD_CALL(data_node_attach);
+TSL_FORWARD_CALL(data_node_detach);
+
+TSL_FORWARD_CALL(chunk_set_default_data_node);
+TSL_FORWARD_CALL(chunk_get_relstats);
+TSL_FORWARD_CALL(chunk_get_colstats);
+
+TSL_FORWARD_CALL(timescaledb_fdw_handler);
+TSL_FORWARD_CALL(timescaledb_fdw_validator);
+TSL_FORWARD_CALL(remote_txn_id_in);
+TSL_FORWARD_CALL(remote_txn_id_out);
+TSL_FORWARD_CALL(remote_txn_heal_data_node);
+TSL_FORWARD_CALL(remote_connection_cache_show);
+TSL_FORWARD_CALL(dist_remote_hypertable_info);
+TSL_FORWARD_CALL(distributed_exec);
+TSL_FORWARD_CALL(hypertable_distributed_set_replication_factor);
 
+TS_FUNCTION_INFO_V1(ts_dist_set_id);
 Datum
 ts_dist_set_id(PG_FUNCTION_ARGS)
 {
 	PG_RETURN_BOOL(ts_cm_functions->set_distributed_id(PG_GETARG_DATUM(0)));
 }
 
+TS_FUNCTION_INFO_V1(ts_dist_set_peer_id);
 Datum
 ts_dist_set_peer_id(PG_FUNCTION_ARGS)
 {
 	ts_cm_functions->set_distributed_peer_id(PG_GETARG_DATUM(0));
 	PG_RETURN_VOID();
 }
 
-Datum
-ts_dist_remote_hypertable_info(PG_FUNCTION_ARGS)
-{
-	PG_RETURN_DATUM(ts_cm_functions->remote_hypertable_info(fcinfo));
-}
-
+TS_FUNCTION_INFO_V1(ts_dist_validate_as_data_node);
 Datum
 ts_dist_validate_as_data_node(PG_FUNCTION_ARGS)
 {
 	ts_cm_functions->validate_as_data_node();
 	PG_RETURN_VOID();
 }
 
-Datum
-ts_distributed_exec(PG_FUNCTION_ARGS)
-{
-	ts_cm_functions->distributed_exec(fcinfo);
-	PG_RETURN_VOID();
-}
```

Comment:
This is different from the others (it uses `PG_RETURN_VOID`) but I agree that it should actually just forward the return value from `distributed_exec`.